lib/src/expand_text.dart

```diff
@@ -123,18 +125,30 @@ class _ExpandTextState extends State<ExpandText>
     });
   }
 
+  Widget buildText() {
+    return Text(
+      widget.text,
+      textAlign: widget.textAlign,
+      overflow: widget.textOverflow,
+      style: widget.style,
+      maxLines: getMaxLines(),
+    );
+  }
+
+  int getMaxLines() {
+    if (widget.textOverflow == TextOverflow.ellipsis) {
+      return _isExpanded ? 2^64 : widget.maxLength;
+    }
+
+    return _isExpanded ? null : widget.maxLength;
+  }
```

Comment:
I wanted to know what's the purpose of the `getMaxLines` method.
Could you explain it to me?

Reply:
I was having trouble getting it to expand fully when overflow was Ellipsis.

I'd recommend you comment of the if statement and set the `textOverflow` to Ellipsis to see what I mean.

It could be related to this comment in the docs.
```
If maxLength is null, but there is an ambient [DefaultTextStyle] that specifies
an explicit number for its [DefaultTextStyle.maxLines], then the
[DefaultTextStyle] value will take precedence.
```

Reply:
OK, that makes sense.
I've improved the readability of this chunk of code & approved the PR.

Thank you very much for taking the time to fix this!